scripts/tools/zap/generate.py

```diff
@@ -58,18 +58,37 @@ def getDirPath(name):
     return fullpath
 
 
+def detectZclFile(zapFile):
```

Comment:
Need to fix `scripts/tools/zap/run_zaptool.sh` too.  Otherwise the next person who uses that (which is the normal thing to use) to edit the .zap file will blow away the changes this PR made...

Reply:
Do we have 2 scripts doing the same thing? Should we have 2 scripts?

Reply:
File content detection using shell is somewhat painful. I can hard-code strings for sure, but generally this seems off.

Reply:
> Do we have 2 scripts doing the same thing?

We have three scripts:

1. `generate.py` -- Take the input .zap and the XML, generate things based on templates.  This is used to codegen.
2. `convert.py` -- Take the input .zap and the XML, fix up the information in the .zap that is redundant with the XML (e.g. the command/attribute names) to match the XML.  This is used to batch-do the work that would happen anyway if the .zap files got edited so as to reduce the noise from the edits.
3. `run_zaptool.sh` -- Take the input .zap and the XML, run the ZAP UI tool on the .zap so that it can be modified and saved.  This is what's used to change things like which commands/attribues are enabled, attribute default values, attribute storage types, endpoint configuration, etc, etc.  Generally after running this and saving you would run `generate.py` to generated the updated code.

Reply:
Thanks!

Updated the other 2 scripts and validated that they run.